ui/app/pages/swaps/swaps.util.js

```diff
@@ -365,11 +365,12 @@ export function quotesToRenderableData (quotes, gasPrice, conversionRate, curren
 }
 
 export function getSwapsTokensReceivedFromTxMeta (tokenSymbol, txMeta, tokenAddress, accountAddress, tokenDecimals) {
+  const txReceipt = txMeta?.txReceipt
   if (tokenSymbol === 'ETH') {
     if (!txMeta || !txMeta.postTxBalance || !txMeta.preTxBalance) {
       return null
     }
-    const gasCost = calcGasTotal(txMeta.txParams.gas, txMeta.txParams.gasPrice)
+    const gasCost = calcGasTotal(txReceipt.gasUsed, txMeta.txParams.gasPrice)
```

Comment:
If `txReceipt` may not exist (per line 368), shouldn't we fall back to `txMeta.txParams.gas` or something?

Reply:
Or return null a couple lines up from here if the receipt is missing.

Reply:
Done in e54f88992